src/services/factrak.ts

```diff
@@ -107,12 +107,12 @@ export class FactrakService {
 
   async getProfessorRatings(
     professorID: number,
-    courseID?: number
-  ): Promise<APIResponse<ModelsFactrakSurveyAvgRatings>> {
+    params?: {
+      courseID?: number,
+      metric?: string
```

Comment:
On second thought, we should specify which `metric`s are valid with [string literal types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-1-8.html#string-literal-types) and use a more descriptive type instead of `string`.

Reply:
I dealt with this in the go code but this could be better and would remove the need for the check in wso-go

Reply:
It doesn't remove the need in golang, as an attacker can easily use curl to send a HTTP request without the typescript check. 

The type here is for preventing typos in our react code (as typescript compiler finds it out) and for IDE auto-completions

Reply:
Resolved in #12